server/routerlicious/packages/services-shared/src/startupChecker.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * This class is used to check if a service/functionality has started up and is ready for use.
+ * This is a singleton class so that it need not be passed as a resource to all services where it is needed.
+ * It is used in the health check endpoint - /startup - to check if the service is ready for use.
+ * @internal
+ */
+export class StartupChecker {
+	private static instance: StartupChecker;
+	private isReady: boolean = false;
+
+	private constructor() {}
+
+	public static getInstance(): StartupChecker {
```

Comment:
(nit): this could be a "getter" which _might_ look cleaner, in the same way that Lumberjack does it.

```ts
private static _instance: StartupChecker | undefined;
public static get instance(): StartupChecker {
  // ...
}

//...

StartupChecker.instance.isReady()
```